Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m

```diff
@@ -178,6 +252,176 @@ - (void)loadForDisplay
+                                                NSForegroundColorAttributeName : [UIColor lightGrayColor],
+                                            }
+                                         referenceView:self];
+        attributedText = [attributedText rtlSafeAppend:@" "
+                                            attributes:@{
+                                                NSFontAttributeName : self.dateHeaderDateFont,
+                                            }
+                                         referenceView:self];
+        attributedText = [attributedText rtlSafeAppend:timeString
+                                            attributes:@{
+                                                NSFontAttributeName : self.dateHeaderTimeFont,
+                                                NSForegroundColorAttributeName : [UIColor lightGrayColor],
+                                            }
+                                         referenceView:self];
+
+        self.dateHeaderLabel.attributedText = attributedText;
+        self.dateHeaderLabel.hidden = NO;
+
+        self.dateHeaderConstraints = @[
+            // Date headers should be visually centered within the conversation view,
+            // so they need to extend outside the cell's boundaries.
+            [self.dateHeaderLabel autoSetDimension:ALDimensionWidth toSize:contentWidth],
+            (self.isIncoming ? [self.dateHeaderLabel autoPinEdgeToSuperviewEdge:ALEdgeLeading]
+                             : [self.dateHeaderLabel autoPinEdgeToSuperviewEdge:ALEdgeTrailing]),
+            [self.dateHeaderLabel autoSetDimension:ALDimensionHeight toSize:self.dateHeaderHeight],
+        ];
+    } else {
+        self.dateHeaderLabel.hidden = YES;
+        self.dateHeaderConstraints = @[
+            [self.dateHeaderLabel autoSetDimension:ALDimensionHeight toSize:0],
+        ];
+    }
+}
+
+- (CGFloat)footerHeight
+{
+    BOOL showFooter = NO;
+
+    TSMessage *message = (TSMessage *)self.viewItem.interaction;
+    BOOL hasExpirationTimer = message.shouldStartExpireTimer;
+
+    if (hasExpirationTimer) {
+        showFooter = YES;
+    } else if (!self.isIncoming) {
+        showFooter = !self.viewItem.shouldHideRecipientStatus;
+    } else if (self.viewItem.isGroupThread) {
+        showFooter = YES;
+    } else {
+        showFooter = NO;
+    }
+
+    return (showFooter ? MAX(kExpirationTimerViewSize,
+                             self.footerLabel.font.lineHeight)
+            : 0.f);
+}
+
+- (void)updateFooter
+{
+    OWSAssert(self.viewItem.interaction.interactionType == OWSInteractionType_IncomingMessage
+        || self.viewItem.interaction.interactionType == OWSInteractionType_OutgoingMessage);
+
+    TSMessage *message = (TSMessage *)self.viewItem.interaction;
+    BOOL hasExpirationTimer = message.shouldStartExpireTimer;
+    NSAttributedString *attributedText = nil;
+    if (!self.isIncoming) {
```

Comment:
TIOLI: dare we invent an isOutgoing getter?

Reply:
You're right, this redundancy will be more readable.